0_From_Code_First_To_First_Principles/1_code_first/assignment/src/controller/user.controller.ts

```diff
@@ -0,0 +1,188 @@
+import express, { Request, Response } from "express";
+import prisma from "../conf/prismaClient";
+import usernameAlreadyTakenException from "../exceptions/userNameAlreadyTakenException";
+import { randomUUID } from "crypto";
+import { ICreateUserResponse } from "./CreateUser/ICreateUserResponse";
+import { ICreateUserRequest } from "./CreateUser/ICreateUserRequest";
+import validationError from "../exceptions/validationError";
+import { HttpStatusCode } from "../utils/enums/statusCode";
+import { User } from "@prisma/client";
+import emailAlreadyInUseException from "../exceptions/emailAlreadyInUseException";
+import { IEditUserRequest } from "./EditUser/IEditUserRequest";
+import { IEditUserResponse } from "./EditUser/IEditUserResponse";
+import { userNotFoundException } from "../exceptions/userNotFoundException";
+import { IGetUserRequestQuery } from "./GetUser/IGetUserRequest";
+import { IGetUserResponse } from "./GetUser/IGetUserResponse";
+
+export class UserController {
+  constructor() {}
+
+  static async createUser(
```

Comment:
Love the strict typing on the request objects. Didn't even know you could do that with express. Nicely done!